Extensions/TileMap/pixi-tilemap-helper.js

```diff
@@ -0,0 +1,261 @@
+      }
+    })
+    const newTileset = {
+      width: tex.width,
+      height: tex.height,
+      tileWidth,
+      tileHeight,
+      texture: tex,
+      textureCache,
+      layers: tiledData.layers,
+      tiles,
+      tilecount,
+      // these scales are used as multipliers on some absolute pixel values from tiled that might be on a resized atlas
+      // they help to adopt to changed dimentions when rendering in updatePIXITileMap
+      margin,
+      spacing,
+    }
+    onLoad(newTileset)
+    loadedTileSets[requestedTileSetId] = newTileset
+  }
+  exports.createTileSetResource = createTileSetResource
+
+  /**
+   * Decodes the layer data, which tiled can sometimes store as a compressed base64 string
+   * https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#data
+   */
+  const decodeBase64 = (layer, pako) => {
+    const { data, compression } = layer
+    var index = 4,
+      arr = [],
+      step1 = atob(data)
+        .split('')
+        .map(function (x) {
+          return x.charCodeAt(0)
+        })
+    try {
+      const decodeString = (str, index) =>
+        (str.charCodeAt(index) +
+          (str.charCodeAt(index + 1) << 8) +
+          (str.charCodeAt(index + 2) << 16) +
+          (str.charCodeAt(index + 3) << 24)) >>>
+        0
+      const decodeArray = (arr, index) =>
+        (arr[index] +
+          (arr[index + 1] << 8) +
+          (arr[index + 2] << 16) +
+          (arr[index + 3] << 24)) >>>
+        0
+
+      if (compression === 'zlib') {
+        var binData = new Uint8Array(step1)
+        step1 = pako.inflate(binData)
+        while (index <= step1.length) {
+          arr.push(decodeArray(step1, index - 4))
+          index += 4
+        }
+      } else {
+        while (index <= step1.length) {
+          arr.push(decodeString(step1, index - 4))
+          index += 4
+        }
+      }
+      return arr
+    } catch (error) {
+      console.error(
+        'Failed to decompress and unzip base64 layer.data string',
+        error
+      )
+      return null
+    }
+  }
+
+  /**
+   * Re-renders the tilemap whenever its rendering settings have been changed
+   */
+  exports.updatePIXITileMap = (tileMap, tileSet, render, layerIndex, pako) => {
+    if (!tileMap || !tileSet) return
+    tileMap.clear()
+
+    tileSet.layers.forEach(function (layer, index) {
+      if (render === 'index' && layerIndex !== index) return
+      else if (render === 'visible' && !layer.visible) return
+
+      // todo filter groups
+      if (layer.type === 'objectgroup') {
+        layer.objects.forEach(function (object) {
+          const { gid, x, y, visible } = object
+          if (render === 'visible' && !visible) return
+          if (tileSet.textureCache[gid]) {
+            tileMap.addFrame(
+              tileSet.textureCache[gid],
+              x,
+              y - tileSet.tileHeight
+            )
+          }
+        })
+      } else if (layer.type === 'tilelayer') {
+        let ind = 0
```

Comment:
Use full names for your variables: `index`. Only use `i` for for loops (if you can't use `forEach` or `map`).

Reply:
can I call it tileIndex? It betted describes it

Reply:
`tileIndex` way better. No problem for i/j for the two nested loops as it's fairly obvious in a for loop what you iterate on. It's less the case here so `tileIndex` is better 👍

Reply:
I ended up calling it  tileSlotIndex, since that is even more descriptive in my mind